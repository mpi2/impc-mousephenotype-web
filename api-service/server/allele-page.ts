import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
