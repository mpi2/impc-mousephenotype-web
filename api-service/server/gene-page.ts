import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
