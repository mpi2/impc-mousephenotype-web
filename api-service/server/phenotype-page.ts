import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { PhenotypeSummary } from "@/models/phenotype";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchPhenotypeSummary(
  phenotypeId: string,
): Promise<PhenotypeSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotype_summary?phenotypeId=${phenotypeId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<PhenotypeSummary>(
        `/api/v1/phenotypes/${phenotypeId}/summary`,
      )
    : fetchURL<PhenotypeSummary>(endpointURL));
}
