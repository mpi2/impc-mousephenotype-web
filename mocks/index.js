export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

