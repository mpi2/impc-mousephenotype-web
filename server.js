const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});