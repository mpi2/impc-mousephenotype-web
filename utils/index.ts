import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
