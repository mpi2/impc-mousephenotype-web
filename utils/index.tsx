import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
