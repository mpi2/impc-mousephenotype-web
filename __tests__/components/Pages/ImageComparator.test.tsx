import ImagesCompare from "@/app/genes/[pid]/images/[parameterStableId]/image-viewer-page";
import { API_URL, renderWithClient } from "../../utils";
import { server } from "../../../mocks/server";
import { rest } from "msw";
import dbn1MutantImages from "../../../mocks/data/tests/dbn1-xray-mutant-images-data.json";
import dbn1ControlImages from "../../../mocks/data/tests/dbn1-xray-control-images-data.json";
import dbn1LaczMutantImages from "../../../mocks/data/tests/dbn1-lacz-mutant-images-data.json";
import dbn1LaczControlImages from "../../../mocks/data/tests/dbn1-lacz-control-images-data.json";
import { screen, waitFor } from "@testing-library/react";
import { useParams, useSearchParams } from "next/navigation";
import userEvent from "@testing-library/user-event";

jest.mock("next/navigation", () => {
  const originalModule = jest.requireActual("next/navigation");
  const { useRouter } = jest.requireActual("next-router-mock");
  return {
    __esModule: true,
    ...originalModule,
    useRouter: jest.fn().mockImplementation(useRouter),
    usePathname: jest.fn(),
    useParams: jest.fn().mockImplementation(() => ({})),
    useSearchParams: jest.fn().mockImplementation(() => new URLSearchParams()),
  };
});

describe("Image comparator page", () => {
  it("renders correctly", async () => {
    const user = userEvent.setup();
    // @ts-ignore
    useParams.mockImplementation(() => ({
      pid: "MGI:1931838",
      parameterStableId: "IMPC_XRY_048_001",
    }));
    server.use(
      rest.get(
        `${API_URL}/api/v1/images/find_by_mgi_and_stable_id`,
        (req, res, ctx) => {
          const paramStableId = req.url.searchParams.get("parameterStableId");
          const mgiGeneAccessionId = req.url.searchParams.get("pid");
          if (
            paramStableId === "IMPC_XRY_048_001" &&
            mgiGeneAccessionId === "MGI:1931838"
          ) {
            return res(ctx.json(dbn1MutantImages));
          }
        },
      ),
    );
    server.use(
      rest.get(
        `${API_URL}/api/v1/images/find_by_stable_id_and_sample_id`,
        (req, res, ctx) => {
          const paramStableId = req.url.searchParams.get("parameterStableId");
          const biologicalSampleGroup = req.url.searchParams.get(
            "biologicalSampleGroup",
          );
          if (
            paramStableId === "IMPC_XRY_048_001" &&
            biologicalSampleGroup === "control"
          ) {
            return res(ctx.json(dbn1ControlImages));
          }
        },
      ),
    );
    // await mockRouter.push("/genes/MGI:1931838/images/IMPC_XRY_048_001");
    const { container } = renderWithClient(
      <ImagesCompare
        mutantImagesFromServer={[]}
        controlImagesFromServer={[]}
      />,
    );
    await waitFor(() =>
      expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
        "X-ray / XRay Images Whole Body Lateral Orientation",
      ),
    );
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(61);
    });
    const headers = await screen.findAllByRole("heading", { level: 3 });
    expect(
      headers
        .map((h) => h.textContent)
        .join("|")
        .trim(),
    ).toBe("WT Images (50)|Mutant Images (11)");

    expect(container).toMatchSnapshot();
    await user.click(screen.getByTestId("heterozygote-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(61);
    });
    expect(await screen.findByTestId("selected-image-mutant")).toHaveAttribute(
      "src",
      "//www.ebi.ac.uk/mi/media/omero/webgateway/render_image/1069405/",
    );
    expect(
      screen.getByTestId("active-heterozygote-filter"),
    ).toBeInTheDocument();
    await user.click(screen.getByTestId("homozygote-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(50);
    });
    expect(screen.getByTestId("active-homozygote-filter")).toBeInTheDocument();
    await user.click(screen.getByTestId("hemizygote-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(50);
    });
    expect(screen.getByTestId("active-hemizygote-filter")).toBeInTheDocument();
    await user.click(screen.getByTestId("both-zygs-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(61);
    });
    expect(screen.getByTestId("active-both-zygs-filter")).toBeInTheDocument();
    const allThumbnails = await screen.findAllByTestId("single-image");
    await user.click(allThumbnails[1]);
    expect(await screen.findByTestId("selected-image-WT")).toHaveAttribute(
      "src",
      "//www.ebi.ac.uk/mi/media/omero/webgateway/render_image/1438446/",
    );
  });

  it("filters images by anatomy query term", async () => {
    const user = userEvent.setup();
    // @ts-ignore
    useParams.mockImplementation(() => ({
      pid: "MGI:1931838",
      parameterStableId: "IMPC_ALZ_075_001",
    }));
    // @ts-ignore
    useSearchParams.mockImplementation(
      () => new URLSearchParams({ anatomyTerm: "adrenal gland" }),
    );
    server.use(
      rest.get(
        `${API_URL}/api/v1/images/find_by_mgi_and_stable_id`,
        (req, res, ctx) => {
          const paramStableId = req.url.searchParams.get("parameterStableId");
          const mgiGeneAccessionId = req.url.searchParams.get("pid");
          if (
            paramStableId === "IMPC_ALZ_075_001" &&
            mgiGeneAccessionId === "MGI:1931838"
          ) {
            return res(ctx.json(dbn1LaczMutantImages));
          }
        },
      ),
    );
    server.use(
      rest.get(
        `${API_URL}/api/v1/images/find_by_stable_id_and_sample_id`,
        (req, res, ctx) => {
          const paramStableId = req.url.searchParams.get("parameterStableId");
          const biologicalSampleGroup = req.url.searchParams.get(
            "biologicalSampleGroup",
          );
          if (
            paramStableId === "IMPC_ALZ_075_001" &&
            biologicalSampleGroup === "control"
          ) {
            return res(ctx.json(dbn1LaczControlImages));
          }
        },
      ),
    );
    // await mockRouter.push("/genes/MGI:1931838/images/IMPC_XRY_048_001");
    renderWithClient(
      <ImagesCompare
        mutantImagesFromServer={[]}
        controlImagesFromServer={[]}
      />,
    );
    await waitFor(() =>
      expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
        "Adult LacZ / LacZ Images Section",
      ),
    );
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(5);
    });
    const headers = await screen.findAllByRole("heading", { level: 3 });
    expect(
      headers
        .map((h) => h.textContent)
        .join("|")
        .trim(),
    ).toBe("WT Images (3)|Mutant Images (2)");
    expect(
      await screen.findByTestId("active-anatomy-filter"),
    ).toBeInTheDocument();
    expect(
      await screen.findByTestId("control-controlCenterFilter"),
    ).toHaveValue("JAX");
    expect(await screen.findByTestId("control-mutantCenterFilter")).toHaveValue(
      "CCP-IMG",
    );
    await user.click(screen.getByTestId("female-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(3);
    });
    expect(screen.getByTestId("active-female-filter")).toBeInTheDocument();
    await user.click(screen.getByTestId("male-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(2);
    });
    expect(screen.getByTestId("active-male-filter")).toBeInTheDocument();
    await user.click(screen.getByTestId("both-sexes-filter"));
    await waitFor(async () => {
      const imgs = await screen.findAllByTestId("single-image");
      return expect(imgs.length).toEqual(5);
    });
    expect(screen.getByTestId("active-both-sexes-filter")).toBeInTheDocument();
  });
});
