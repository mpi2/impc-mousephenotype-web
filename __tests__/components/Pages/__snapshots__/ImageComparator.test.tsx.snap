// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image comparator page renders correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="pb-4 pt-5 container container"
    >
      <div
        class="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          class="tabs"
        >
          <a
            class="tab__active"
            href="/search"
          >
            Genes
          </a>
          <a
            class="tab"
            href="/search?type=pheno"
          >
            Phenotypes
          </a>
          <a
            class="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          class="inputCont"
        >
          <input
            class="input"
            placeholder="Search for a gene..."
            title="main search box"
            type="text"
            value=""
          />
          <button
            aria-describedby="svg-inline--fa-title-search-icon"
            class="searchBtn"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-search-icon"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-search-icon"
              >
                Search button
              </title>
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page container"
  >
    <div
      class="card"
    >
      <div
        class="subheading"
      >
        <span
          class="subheadingSection primary"
        >
          <a
            class="mb-3"
            href="/genes/MGI:96853#images"
            style="text-transform: none; font-weight: normal; letter-spacing: normal; font-size: 1.15rem;"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left "
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
              Go Back to
             
            <i>
              <span
                aria-busy="true"
                aria-live="polite"
              >
                <span
                  class="react-loading-skeleton"
                  style="width: 50px;"
                >
                  ‌
                </span>
              </span>
            </i>
          </a>
        </span>
      </div>
      <p
        class="subheading mt-2"
      >
        Images
      </p>
      <h1
        class="mb-4 mt-2"
        style="display: flex; gap: 1rem;"
      >
        <strong>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="display: block; width: 300px;"
            >
              ‌
            </span>
          </span>
        </strong>
         / 
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="display: block; width: 300px;"
          >
            ‌
          </span>
        </span>
      </h1>
      <div>
        <div
          class="row"
        >
          <div
            class="col-sm-6"
          >
            <div
              class="headerContainer"
            >
              <h3
                style="margin-bottom: 0px;"
              >
                WT Images (
                )
              </h3>
              <div
                data-test-id="filterbox-Center"
              >
                <label
                  class="grey"
                  for="controlCenterFilter"
                  style="margin-right: 0.5rem;"
                >
                  Center
                  :
                </label>
                <input
                  class="bg-white form-control"
                  id="controlCenterFilter"
                  placeholder="Search"
                  style="display: inline-block; width: 145px;"
                  type="text"
                  value="IMPC"
                />
              </div>
            </div>
            <div
              class="col-12"
            >
              <div
                class="ratio ratio-16x9 imageContainer"
              >
                <span
                  aria-busy="true"
                  aria-live="polite"
                  class="flex-1"
                >
                  <span
                    class="react-loading-skeleton"
                    style="flex: 1; height: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
              <div
                class="imageInfo"
              />
            </div>
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="headerContainer"
            >
              <h3
                style="margin-bottom: 0px;"
              >
                Mutant Images (
                )
              </h3>
              <div
                data-test-id="filterbox-Center"
              >
                <label
                  class="grey"
                  for="mutantCenterFilter"
                  style="margin-right: 0.5rem;"
                >
                  Center
                  :
                </label>
                <input
                  class="bg-white form-control"
                  id="mutantCenterFilter"
                  placeholder="Search"
                  style="display: inline-block; width: 145px;"
                  type="text"
                  value="IMPC"
                />
              </div>
            </div>
            <div
              class="col-12"
            >
              <div
                class="ratio ratio-16x9 imageContainer"
              >
                <span
                  aria-busy="true"
                  aria-live="polite"
                  class="flex-1"
                >
                  <span
                    class="react-loading-skeleton"
                    style="flex: 1; height: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </div>
              <div
                class="imageInfo"
              />
            </div>
          </div>
        </div>
        <div
          class="mt-3 mb-3 row"
        >
          <div
            class="col-12"
          >
            <div
              class="legendsContainer"
            >
              <span>
                Zygosity indicators
              </span>
              <span
                class="legendWrapper"
              >
                <div
                  class="legendBackground"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle "
                    data-icon="circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="color: rgb(255, 255, 255);"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Wildtype
              </span>
              <span
                class="legendWrapper"
              >
                <div
                  class="legendBackground"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle "
                    data-icon="circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="color: rgb(136, 204, 238);"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Heterozygote
              </span>
              <span
                class="legendWrapper"
              >
                <div
                  class="legendBackground"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle "
                    data-icon="circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="color: rgb(221, 204, 119);"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Homozygote
              </span>
              <span
                class="legendWrapper"
              >
                <div
                  class="legendBackground"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle "
                    data-icon="circle"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="color: rgb(204, 102, 119);"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Hemizygote
              </span>
            </div>
          </div>
          <div
            class="col-12"
          >
            <div
              class="filtersWrapper"
            >
              <div
                class="column"
              >
                Filter by:
                <div
                  class="filter"
                >
                  <strong>
                    Sex:
                  </strong>
                  <span
                    class="badge active badge rounded-pill bg-badge-secondary"
                  >
                    All
                     
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-mars-and-venus "
                      data-icon="mars-and-venus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M337.8 14.8C341.5 5.8 350.3 0 360 0H472c13.3 0 24 10.7 24 24V136c0 9.7-5.8 18.5-14.8 22.2s-19.3 1.7-26.2-5.2l-39-39-24.7 24.7C407 163.3 416 192.6 416 224c0 80.2-59.1 146.7-136.1 158.2c0 .6 .1 1.2 .1 1.8v.4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .4 .3 .4 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3 .3h24c13.3 0 24 10.7 24 24s-10.7 24-24 24H280v.2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .2 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 .1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0l-24 0-24 0v0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1V486 486v-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1V485 485v-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1V484v-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1V483v-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1-.1V481v-.1-.1-.1-.1-.1-.1-.1-.1V480v-.1-.1-.1-.1-.1-.1-.1V479v-.1-.1-.1-.1-.1-.1-.1V478v-.1-.1-.1-.1-.1-.1V477v-.1-.1-.1-.1-.1-.1V476v-.1-.1-.1-.1-.1-.1V475v-.1-.2-.2-.2-.2-.2V474v-.2-.2-.2-.2-.2V473v-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2V470v-.2-.2-.2-.2-.2V469v-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2V467v-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2V463v-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2-.2V459v-.2-.2-.2-.2-.2-.2-.2-.2V457v-.2-.2-.2-.2V456H208c-13.3 0-24-10.7-24-24s10.7-24 24-24h24v-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3V403v-.3-.3V402v-.3-.3V401v-.3-.3V400v-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.3-.4-.3-.4-.4-.4-.4V393v-.4-.4-.4-.4-.4-.4-.4-.4-.4-.4-.4-.4-.4V388v-.4-.4-.4-.4-.4-.4-.4-.4-.4-.4V384c0-.6 0-1.2 .1-1.8C155.1 370.7 96 304.2 96 224c0-88.4 71.6-160 160-160c39.6 0 75.9 14.4 103.8 38.2L382.1 80 343 41c-6.9-6.9-8.9-17.2-5.2-26.2zM448 48l0 0h0v0zM256 488h24c0 13.3-10.7 24-24 24s-24-10.7-24-24h24zm96-264a96 96 0 1 0 -192 0 96 96 0 1 0 192 0z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <span
                    class="badge badge rounded-pill bg-badge-secondary"
                  >
                    Female
                     
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-venus "
                      data-icon="venus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 384 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M80 176a112 112 0 1 1 224 0A112 112 0 1 1 80 176zM224 349.1c81.9-15 144-86.8 144-173.1C368 78.8 289.2 0 192 0S16 78.8 16 176c0 86.3 62.1 158.1 144 173.1V384H128c-17.7 0-32 14.3-32 32s14.3 32 32 32h32v32c0 17.7 14.3 32 32 32s32-14.3 32-32V448h32c17.7 0 32-14.3 32-32s-14.3-32-32-32H224V349.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <span
                    class="badge badge rounded-pill bg-badge-secondary"
                  >
                    Male
                     
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-mars "
                      data-icon="mars"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M289.8 46.8c3.7-9 12.5-14.8 22.2-14.8H424c13.3 0 24 10.7 24 24V168c0 9.7-5.8 18.5-14.8 22.2s-19.3 1.7-26.2-5.2l-33.4-33.4L321 204.2c19.5 28.4 31 62.7 31 99.8c0 97.2-78.8 176-176 176S0 401.2 0 304s78.8-176 176-176c37 0 71.4 11.4 99.8 31l52.6-52.6L295 73c-6.9-6.9-8.9-17.2-5.2-26.2zM400 80l0 0h0v0zM176 416a112 112 0 1 0 0-224 112 112 0 1 0 0 224z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="column mutantFilter"
              >
                <div
                  class="filter"
                >
                  <strong>
                    Mutant zygosity:
                  </strong>
                  <span
                    class="badge active badge rounded-pill bg-badge-secondary"
                  >
                    All
                     
                  </span>
                  <span
                    class="badge badge rounded-pill bg-badge-secondary"
                  >
                    Heterozygote
                     
                  </span>
                  <span
                    class="badge badge rounded-pill bg-badge-secondary"
                  >
                    Homozygote
                     
                  </span>
                  <span
                    class="badge badge rounded-pill bg-badge-secondary"
                  >
                    Hemizygote
                     
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-sm-6"
          >
            <div
              class="images row"
            />
          </div>
          <div
            class="col-sm-6"
          >
            <div
              class="images row"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
